GSA/COM API/Python/GSA2Compos.py

```diff
@@ -0,0 +1,202 @@
+"""
+Program for linking GSA with Compos for composite beam design
+This script is intended to
+1. open GSA model using gsapy
+2. extract member sections and lengths based on first saved list
+   - write to member_list
+3. extract defined profiles and write to sections_dictionary
+4. write compos_file.csv file using the member_list
+5. open compos_file.csv in Compos, design the members &
+   extract the new sections to compos_results
+6. reopen the GSA model,
+   use compos_results and sections_dictionary to update the members
+"""
+
+import win32com.client  # https://pypi.org/project/pywin32/
+from datetime import datetime
+from tempfile import gettempdir
+from gsapy import GSA
+
+# gsapy references
+# https://gitlab.arup.com/david.dekoning/gsapy/-/blob/master/docs/readme.rst
+# https://gitlab.arup.com/hongkong-rd/JupyterSamples/blob/master/pyGSA/pyGSA_gsapy.ipynb
+
+
+dead_load = 3.0  # kN/m^2
+live_load = 5.0  # kN/m^2
+# add additional variables as necessary to control what is added to Compos
+
+
+def gsa2compos2gsa():
+    global dead_load
+    global live_load
+
+    # get GSA data
+    print("Opening GSA model")
+    gsa_model = GSA(r'C:\Temp\Composite test.gwb', version='10.1')
+
+    # gets the members from the first saved list
+    search_list = \
+        str(gsa_model.get_all_saved_lists()[0]).split("\t")[4].split(" ")
+    member_list = []
+    for i in search_list:  # create list of members to be designed
+        member = str(gsa_model.get_members(int(i)))
+        member_section = int(member.split("\t")[6])
+        member_list.append([int(i), gsa_model.get_section_info(member_section),
+                            gsa_model.get_member_length(int(i))])
+    sections_dictionary = {}
```

Comment:
```python
from gsapy.modules import Section

sections_dictionary = gsa_model.get(Section)
```
This creates a dictionary where the keys are the integer ids and the values are Section objects

Reply:
but I see that you are using the section description as the key, so it needs something different:

```python
sections_dictionary = {s.desc:s for s in gsa_model.get(Section).values}
```
or
```python
sections_dictionary = {gsa_model.get_section_info(s.index):s for s in gsa_model.get(Section).values}
```
you could also expand it as
```python
sections = gsa_model.get(Section) # this gives you the basic dict of sections (keys are the index)
sections_dictionary = {gsa_model.get_section_info(s.index) for s in sections}
```
https://www.geeksforgeeks.org/python-dictionary-comprehension/

Reply:
@daviddekoning this option doesn't work
```
    sections_dictionary = {}
    for i in range(1, 93):  # create sections dictionary:  needs to match model
        sections_dictionary[gsa_model.get_section_info(i)] = i
```
is clunky but gives a dictionary in the form _{'UB127x76x13':1 etc}_, while 
`sections_dictionary4 = {s.desc:s for s in gsa_model.get(Section).values()}`
is elegant but produces something very different.
Is there an elegant way to create the dictionary in the required form?

Reply:
you could say `s.desc.split("%")[2]:s for si in ....`, but what I did in my suggested changes is just use the **full** section name, since they are consistent across GSA and Compos. I just had to replace the %s with spaces:

`sections_dictionary = {s.desc.replace("%", " "): i for i, s in sections.items()}`